spec/unit/typesafe/config/tokenizer_spec.rb

```diff
@@ -0,0 +1,696 @@
+require 'spec_helper'
+require 'hocon'
+require 'test_utils'
+require 'pp'
+
+
+describe Hocon::Impl::Tokenizer do
+  Tokens = Hocon::Impl::Tokens
+
+  shared_examples_for "token_matching" do
+    it "should match the tokenized string to the list of expected tokens" do
+      tokenized_from_string = TestUtils.tokenize_as_list(test_string)
+
+      # Add START and EOF tokens
+      wrapped_tokens = TestUtils.wrap_tokens(expected_tokens)
+
+      # puts "\n\n===Expected String: #{test_string}"
+      # PP.pp(tokenized_from_string)
+
+      # Compare the two lists of tokens
+      expect(tokenized_from_string).to eq(wrapped_tokens)
+    end
+  end
+
+  shared_examples_for "strings_with_problems" do
+    it "should find a problem when tokenizing" do
+      token_list = TestUtils.tokenize_as_list(test_string)
+      expect(token_list.map { |token| Tokens.problem?(token) }).to include(true)
+    end
+  end
+
+  ####################
+  # Whitespace
```

Comment:
Something to consider: you can nest contexts in rspec, so I think it might be worth nesting each of these sections in a higher-level context. So for example, in this section, you could put all of these contexts into a higher level `"tokenizing whitespace"` context.